src/shared/filesystem.ts

```diff
@@ -18,7 +20,52 @@ export interface Stats extends fs.Stats {
 // functions
 export const access = promisify(fs.access)
 
-export const mkdir = promisify(fs.mkdir)
+const _mkdir = promisify(fs.mkdir)
+interface ErrorWithCode {
+    code?: string
+}
+
+export async function mkdir(
```

Comment:
We currently keep thin wrappers in `filesystem.ts`, and custom logic/helpers in `filesystemUtilities.ts`.

Moving forward, we can probably get rid of `filesystem.ts`. But I think these methods belong in `filesystemUtilities.ts`.

Reply:
That would create a cyclical dependency. This is an implementation detail of our enhanced `mkdir`